relay-server/src/metrics_extraction/transactions.rs

```diff
@@ -18,6 +18,19 @@ pub struct TransactionMetricsConfig {
     extract_custom_tags: BTreeSet<String>,
 }
 
+#[cfg(feature = "processing")]
+const METRIC_NAME_PREFIX: &str = "sentry.transactions";
+
+/// Generate a transaction-related metric name
+#[cfg(feature = "processing")]
+fn metric_name(parts: &[&str]) -> String {
+    let mut name = METRIC_NAME_PREFIX.to_owned();
+    for part in parts {
+        name = format!("{}.{}", name, part);
```

Comment:
```suggestion
        write!(".{}", part);
```

this may need some bogus error-handling because it goes through io::Write, but it prevents string allocations